src/plugins/saved_objects_management/public/management_section/objects_table/components/table.tsx

```diff
@@ -231,9 +238,19 @@ export class Table extends PureComponent<TableProps, TableState> {
           if (!canGoInApp) {
             return <EuiText size="s">{title || getDefaultTitle(object)}</EuiText>;
           }
-          return (
-            <EuiLink href={basePath.prepend(path)}>{title || getDefaultTitle(object)}</EuiLink>
-          );
+          let inAppUrl = basePath.prepend(path);
+          if (object.workspaces?.length) {
+            if (currentWorkspaceId) {
+              inAppUrl = formatUrlWithWorkspaceId(path, currentWorkspaceId, basePath);
+            } else {
+              // first workspace user have permission
```

Comment:
`first` -> `filter`?

Reply:
Oh I see the point, it just need to find the first workspace that matches, maybe can simply use `object.workspaces.find`

Reply:
Thanks, updated to `find` method.

Reply:
Still can see `object.workspaces.filter`, am I looking at the right version?

Reply:
ah, seem have code merge issue when rebase from main. let me redo the change.